app/controllers/application_controller.rb

```diff
@@ -5,6 +5,148 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "password_security"
+  end
+### INDEX ###
+  get '/' do 
+    erb :index
+  end
+
+### SIGNUP ###
+  get '/signup' do 
+    if Helpers.is_logged_in?(session)
+      redirect "/tweets"
+    else
+      erb :'/users/create_user'
+    end
+  end
+
+  post '/signup' do 
+    @user = User.new(username: params["username"], email: params["email"], password: params["password"])
+    if !@user.username.empty? && !@user.email.empty? && @user.password != nil
+      @user.save
+      session[:user_session_id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/signup"
+    end
+  end
+
+### LOGIN ###
+  get '/login' do
+    if Helpers.is_logged_in?(session)
+      redirect "/tweets"
+    else
+      erb :'/users/login'
+    end
+  end
+
+  post '/login' do 
+    @user = User.find_by(:username => params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_session_id] = @user.id
+      redirect "/tweets"
+    else
+      redirect "/login"
+    end
+  end
+
+### LOGOUT ###
+  get '/logout' do
+    if Helpers.is_logged_in?(session)
+      session.clear
+      redirect "/login"
+    else
+      redirect "/"
+    end
+  end
+
+  ### SHOW ALL TWEETS OF SPECIFIC USER ###
+  get '/users/:slug' do 
+    @user = User.find_by_slug(params[:slug])
+    erb :'users/show_tweets'
+  end
+
+
+### CREATE NEW TWEET ###
+  get '/tweets/new' do
+    if Helpers.is_logged_in?(session)
+      erb :'/tweets/create_tweet' 
+    else
+      redirect "/login"
+    end
+  end
+
+  post '/tweets/new' do 
+    @tweet = Tweet.new(:content => params["content"], :user_id => params["user_id"])
```

Comment:
Your hypothesis was correct. Try it in Chrome! You can right click a form field, click `inspect`. Then find the hidden field, right click, and select `Edit as HTML`. Additionally, this could be manipulated with a utility such as `curl` or an app like [Postman](https://www.getpostman.com/).